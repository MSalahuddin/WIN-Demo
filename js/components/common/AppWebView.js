import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import { WebView } from 'react-native-webview';

const StyledSafeAreaView = styled.SafeAreaView`
  flex: 1;
`;

const AppWebView = ({ navigation }) => {
  const { url } = navigation.state.params;
  return (
    <StyledSafeAreaView>
      <WebView
        source={{
          uri: url
        }}
        testID="web-view"
        useWebKit
      />
    </StyledSafeAreaView>
  );
};

AppWebView.propTypes = {
  navigation: PropTypes.shape({
    goBack: PropTypes.func.isRequired,
    state: PropTypes.shape({
      params: PropTypes.shape({
        url: PropTypes.string
      })
    })
  }).isRequired
};

export default AppWebView;
