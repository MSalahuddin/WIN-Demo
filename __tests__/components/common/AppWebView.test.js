import React from 'react';
import { render } from 'react-native-testing-library';

import AppWebView from '../../../js/components/common/AppWebView';

describe('Component AppWebView', () => {
  const navigation = {
    goBack: jest.fn(),
    state: {
      params: {
        url: ''
      }
    }
  };

  test('renders correctly', () => {
    const output = render(<AppWebView navigation={navigation} />);
    expect(output).toMatchSnapshot();
  });
});
