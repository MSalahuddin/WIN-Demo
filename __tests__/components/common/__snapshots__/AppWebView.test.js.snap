// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AppWebView renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      testID="web-view"
      useSharedProcessPool={true}
      useWebKit={true}
    />
  </View>
</RCTSafeAreaView>
`;
